tools/utils/http-helpers.js

```diff
@@ -358,6 +369,47 @@ _.extend(exports, {
     } else {
       return body;
     }
-  }
+  },
 
+  getUrlWithResuming(urlOrOptions) {
+    const options = _.isObject(urlOrOptions) ? urlOrOptions : {
+      url: urlOrOptions,
+    };
+
+    const outputStream = new WritableStreamBuffer();
+
+    const MAX_ATTEMPTS = 5;
+    function attempt(triesRemaining) {
+      const headers = _.extend({}, urlOrOptions.headers, {
+        range: `bytes=${outputStream.size()}-`,
+      });
+      try {
+        return httpHelpers.request({
+          outputStream,
+          headers,
+          ...urlOrOptions,
+        });
+      } catch (e) {
+        console.log(e);
+        if (triesRemaining > 0) {
+          console.log(`Request failed, retrying`);
+          return attempt(triesRemaining - 1);
```

Comment:
What if the failure only counted against `triesRemaining` if the download made no progress (or less than some small amount of progress)?